go/kbnm/hostmanifest/whitelist.go

```diff
@@ -0,0 +1,57 @@
+package hostmanifest
+
+import (
+	"encoding/json"
+	"os"
+	"path/filepath"
+)
+
+// whitelistPath is used for installing the whitelist as a JSON into a given
+// path on disk.
+type whitelistPath struct {
+	// Root is the path of the machine-global NativeMessage whitelists.
+	Root string
+	// Home is the path of the NativeMessage whitelists for regular users.
+	Home string
+}
+
+func (w *whitelistPath) path(u User, appID string) string {
+	if u.IsAdmin() {
+		return filepath.Join(u.PrefixPath(), w.Root, appID+".json")
```

Comment:
This still feels like a weird mix of concepts to me. Root users do have home directories in `/root/`, and they might want to use them. Also packaging builds done as non-root users will still want to use the `w.Root` paths, coupled with whatever prefix they're supplying. `PrefixPath` seems to be doing double-duty as a packaging prefix and as the user's home directory.

That said, this is all easy to change whenever we want, so I don't want to hold up the PR on it.